docs/source/graph-construction/gs-processing/gpartition/ec2-clusters.rst

```diff
@@ -0,0 +1,122 @@
+
+After setting up a shared file system, we can keep all partitioned graph data in the shared folder. Then mount the data folder to the ``/path_to_data/`` of each instances in the cluster so that all GraphStorm containers in the cluster can access these partitioned graph data easily.
+
+Create GraphStorm container by mounting the NFS folder
+.......................................................
+In each instance, use the following command to start a GraphStorm Docker container and run it as a backend daemon on cpu.
+
+.. code-block:: shell
+
+    docker run -v /path_to_data/:/data \
+                      -v /dev/shm:/dev/shm \
+                      --network=host \
+                      -d --name test graphstorm:local-cpu service ssh restart
+
+This command mount the shared ``/path_to_data/`` folder to each container's ``/data/`` folder by which GraphStorm codes can access graph data and save training and inference outcomes.
+
+Setup the IP address file and check port status
+----------------------------------------------------------
+
+Collect the IP list
+......................
+The GraphStorm Docker containers use SSH on port ``2222`` to communicate with each other. Users need to collect all IP addresses of the three instances and put them into a text file, e.g., ``/data/ip_list.txt``, which is like:
+
+.. figure:: ../../../../../tutorial/distributed_ips.png
+    :align: center
+
+.. note:: If possible, use **private IP addresses**, insteand of public IP addresses. Public IP addresses may have additional port constraints, which cause communication issues.
+
+Put this file into container's ``/data/`` folder.
+
+Check port
+................
+The GraphStorm Docker container uses port ``2222`` to **ssh** to containers running on other machines without passwords. Please make sure all host instances do not use this port.
+
+Users also need to make sure the port ``2222`` is open for **ssh** commands.
+
+Pick one instance and run the following command to connect to the GraphStorm Docker container.
+
+.. code-block:: bash
+
+    docker container exec -it test /bin/bash
+
+In the container environment, users can check the connectivity with the command ``ssh <ip-in-the-cluster> -o StrictHostKeyChecking=no -p 2222``. Please replace the ``<ip-in-the-cluster>`` with the real IP address from the ``ip_list.txt`` file above, e.g.,
+
+.. code-block:: bash
+
+    ssh 172.38.12.143 -o StrictHostKeyChecking=no -p 2222
+
+If succeeds, you should login to the container in the ``<ip-in-the-cluster>`` instance.
+
+If not, please make sure there is no restriction of exposing port 2222.
+
+For distributed training, users also need to make sure ports under 65536 is open for DistDGL to use.
+
+
+Launch GPartition Job
+----------------------
+
+Now we can ssh into the **leader node** of the EC2 cluster, and start GPartition process with the following command:
```

Comment:
Why not use the **leader node** in the previous examples?

Reply:
The previous example is only only set up for one instance. The only instance there should be the leader node.